smartsim/entity/dbobject.py

```diff
@@ -107,18 +109,24 @@ def _enumerate_devices(self) -> t.List[str]:
         :rtype: list[str]
         """
         devices = []
-        if ":" in self.device and self.devices_per_node > 1:
-            msg = "Cannot set devices_per_node>1 if a device numeral is specified, "
-            msg += f"the device was set to {self.device} and devices_per_node=={self.devices_per_node}"
-            raise ValueError(msg)
-        if self.device in ["CPU", "GPU"] and self.devices_per_node > 1:
+        if self.device in ["GPU"] and self.devices_per_node > 1:
             for device_num in range(self.devices_per_node):
                 devices.append(f"{self.device}:{str(device_num)}")
         else:
             devices = [self.device]
 
         return devices
 
+    def _check_arguments(self):
```

Comment:
i love pulling complex validation out into separate methods but `_check_arguments` feels like it could be unclear *which* arguments are being checked (especially since it has none).

I'd rename this to `_check_init` to make it clear where it's valid to use. I'd also probably pass the inputs instead of allowing the object to maybe be partially initialized.

Reply:
I could see this as being a general sanity checker of the initialization. There's a bit of a tradeoff between cluttering up the constructor and breaking it out into a validator for just devices and number of devices.

As currently coded up (where it only relies on the set properties after initialization is completed), it's unambiguous that the settings are correct. Thoughts @juliaputko @ankona?

Reply:
It looks like `_check_xxx` methods in the constructor follow the pattern of offloading per-field validation to another method. I wouldn't care if they're combined but `_check_parameters` is definitely improper naming.